modules/webbrowser/src/webbrowserclient.cpp

```diff
@@ -82,12 +82,15 @@ WebBrowserClient::WebBrowserClient(ModuleManager& moduleManager,
     });
 }
 
-void WebBrowserClient::setBrowserParent(CefRefPtr<CefBrowser> browser, Processor* parent) {
+WebBrowserClient::BrowserParentHandle WebBrowserClient::setBrowserParent(
+    CefRefPtr<CefBrowser> browser, Processor* parent) {
     CEF_REQUIRE_UI_THREAD();
-    BrowserData bd{parent, new ProcessorCefSynchronizer(parent)};
+    auto processor = std::make_shared<Processor*>(parent);
```

Comment:
What is going on here? How you are allowed to take ownership if this processor? Processors are owned by the processor network. Unless this processor it not in the network this is UB

Reply:
Perhaps a bit unclear, but it is only the pointer to the processor that is kept.
I first thought of calling it dummy or something...
Any suggestion on how to make it more expressive.

An alternative that I thought of was to observe the network, but I thought this was more self-contained

Reply:
I still don't understand, the processor is not in the network? And in that case What do you need it for?

Reply:
Humm looking again, so it is a shared_ptr to a __pointer__ to a processor. Now I'm even more confused, why would you have that?

Reply:
I need a reference to the processor in order for the browser to access it. It has turned out that the Processor might be deleted before the reference to it is removed (OnBeforeBrowserClose). So I solved it by keeping a weak_ptr around. 
I considered other alternatives:
- Processor explicitly removes itself as "parent" upon destruction
- Let the WebbrowserClient observe the network and remove the parent if the processor was removed
- Use a handle that needs to be kept alive as long as the processor is alive

I chose the third option...

Reply:
Ahh so you lock that one before you invalidate? Is that the only place you need to do that? Should that not
Be the same for all property changes?

Reply:
Hmm, yes I guess you could get unlucky... 
Perhaps explicitly removing it upon processor destruction is a better option